### PR TITLE
Implementing NSCopying for instrumentation event

### PR DESCRIPTION
**Pull request by khawkins**
For safety/convention during I/O.

### CODE REVIEW

File: libs/SalesforceAnalytics/SalesforceAnalytics/Classes/Model/SFSDKInstrumentationEvent.m [hunk 1]
```diff
@@ -133,6 +133,56 @@ - (instancetype) initWithJson:(NSData *) jsonRepresentation {
     return self;
 }
 
+#pragma mark - NSCopying
+
+- (instancetype)copyWithZone:(NSZone *)zone {
```
Comment (khawkins) [inline libs/SalesforceAnalytics/SalesforceAnalytics/Classes/Model/SFSDKInstrumentationEvent.m]: This is the only new method here.  The other changes in this file are just moving methods around to group them.

File: libs/SalesforceAnalytics/SalesforceAnalyticsTests/InstrumentationEventBuilderTests.m [hunk 2]
```diff
@@ -151,41 +158,37 @@ - (void) testAutoPopulateStartTime {
...
+
+#pragma mark - Helper methods
+
+- (SFSDKInstrumentationEvent *)standardTestEvent {
+    return [SFSDKInstrumentationEventBuilder buildEventWithBuilderBlock:^(SFSDKInstrumentationEventBuilder *builder) {
         double curTime = 1000 * [[NSDate date] timeIntervalSince1970];
         NSString *eventName = [NSString stringWithFormat:kTestEventName, curTime];
         builder.name = eventName;
         builder.page = [[NSDictionary alloc] init];
+        builder.attributes = @{ @"Name1": @"Value1" };
```
Comment (bhariharan) [inline libs/SalesforceAnalytics/SalesforceAnalyticsTests/InstrumentationEventBuilderTests.m]: Why `attributes`? This is unrelated, I presume?

Comment (khawkins) [inline libs/SalesforceAnalytics/SalesforceAnalyticsTests/InstrumentationEventBuilderTests.m]: Ah, I was adding something for `attributes` just to have it populated, testing some copy scenarios.  This doesn't need to be here.  I'll remove it.
